src/controller/python/chip/ChipStack.py

```diff
@@ -367,7 +393,25 @@ def Call(self, callFunct, timeoutMs: int = None):
             return self.callbackRes
         return res
 
-    def CallAsync(self, callFunct):
+    async def CallAsync(self, callFunct, timeoutMs: int = None):
+        '''Run a Python function on CHIP stack, and wait for the response.
+        This function will post a task on CHIP mainloop and waits for the call response in a asyncio friendly manner.
+        '''
+        loop = asyncio.get_event_loop()
+        future = loop.create_future()
+        callObj = AsyncSimpleCallableHandle(callFunct, loop, future)
+        pythonapi.Py_IncRef(py_object(callObj))
+
+        res = self._ChipStackLib.pychip_DeviceController_PostTaskOnChipThread(
```

Comment:
> If I had to guess self.networkLock is likely old code that needs to just be removed, but for the time being I think for keeping things the same it might be good to add that here

Yeah I came essentially to this conclusion to, but decided to start to stop using it with this PR.

The problem with the lock is that it's blocking code. In a asyncio world, it is paramount that the event loop doesn't gets blocked, otherwise the whole application might stall. This makes any `threading.Lock`, or `threading.Complete` are nogo in (clean) asyncio world.

The question becomes what is the lock exactly protecting? I guess concurrent calls of `Call` and `CallAsync`, somehow, but as the code is I really wonder if that would be a problem without a lock? :thinking:

Reply:
If you are not using it for that reason then I would support having a comment added here, with another TODO comment linking to an issues above the existing lock acquisition code asking what the lock is protecting and if it should be removed now.

Reply:
Ok, created the issue (see https://github.com/project-chip/connectedhomeip/issues/33321) and added the comments.